pkg/config/config.go

```diff
@@ -47,11 +47,10 @@ func (in *Install) DeepCopyInto(out *Install) {
 }
 
 type Registration struct {
-	URL             string `json:"url,omitempty" yaml:"url,omitempty" mapstructure:"url"`
-	CACert          string `json:"ca-cert,omitempty" yaml:"ca-cert,omitempty" mapstructure:"ca-cert"`
-	EmulateTPM      bool   `json:"emulate-tpm,omitempty" yaml:"emulate-tpm,omitempty" mapstructure:"emulate-tpm"`
-	EmulatedTPMSeed int64  `json:"emulated-tpm-seed,omitempty" yaml:"emulated-tpm-seed,omitempty" mapstructure:"emulated-tpm-seed"`
```

Comment:
this structure is nested in the machine registration object, removing the field makes it a breaking change. It's only used for development so this is probably fine but we have to be aware that such change removes a field from a released API.

Reply:
was thinking about setting here the special value `-1` to indicate "create a random seed" and then thought...why not just create it all the time? Why do we have a explicit seed in here? whats the purpose?

So yeah, sent this to run the tests and check, we can still do what I thought on my first implementation, just check for the `-1` and then generate a random seed, while keeping the struct intact, but I dont think it has any value in there :/

Reply:
We even mention on the docs that this is for debugging purpouses only. In fact, re-reading it, it appears as we use a random seed and you cant then set that value to use a fixed one...yikes, not good

`fixed seed to use with 'emulate-tpm': use for debug purposes only`

Reply:
Just to make sure, this seed (or the emulated tpm hash) will get persisted across reboots and upgrades, right?

Reply:
I think the TPMHash derived from this is the one used afterwards...Im not even sure its used afterwards, only in the registration? 

@fgiudici I think you have the most knowledge in this area, is the TPMHash used afterwards for secure communication, or its just used on the first registration only? I cant see it used anywhere else....

Reply:
FYI I deployed a cluster using the random seed mechanism already and could not see any issues, hence me wondering where is the TPMSeed used afterwards. system-agent nor k8s has any concept of the TPMHash so I think its only valid for the registration.

Reply:
I use it (soft-tpm with a fixed seed) regularly to test "create machineInventory first, then run installation" with RPi.

Reply:
yes, but after registration Im not sure where its used. Also that same workflow would work with a random seed, I cant see how would having a fixed VS random seed would give us any advantage here but maybe its becasue I dont know that much about the tpm stuff in the registration